projects/scripts/adi_board.tcl

```diff
@@ -460,7 +487,11 @@ proc ad_xcvrcon {u_xcvr a_xcvr a_jesd {lane_map {}} {link_clk {}} {device_clk {}
   }
 
   if {$tx_or_rx_n == 0} {
-    set xcvr_rx_index [expr ($xcvr_rx_index + $no_of_lanes)]
+    if {$num_of_max_lanes != -1} {
+      set xcvr_rx_index [expr ($xcvr_rx_index + $max_no_of_lanes)]
+    } else {
+      set xcvr_rx_index [expr ($xcvr_rx_index + $no_of_lanes)]
+    }
   }
 
   if {$tx_or_rx_n == 1} {
```

Comment:
Also in case of Tx, xcvr_tx_index  can be increased always with max_no_of_lanes

Reply:
Fixed all the bugs you mentioned